Tests/PackageGraphTests/PackageGraphTests.swift

```diff
@@ -548,7 +548,87 @@ class PackageGraphTests: XCTestCase {
+                        PackageDependencyDescription(name: "Bar", url: "/Bar", requirement: .branch("master")),
+                        PackageDependencyDescription(name: nil, url: "/BizPath", requirement: .exact("1.2.3")),
+                        PackageDependencyDescription(name: nil, url: "/FizPath", requirement: .upToNextMajor(from: "1.1.2")),
+                    ],
+                    targets: [
+                        TargetDescription(name: "Foo", dependencies: ["BarLib", "Biz", "FizLib"]),
+                    ]),
+                Manifest.createV4Manifest(
+                    name: "Bar",
+                    path: "/Bar",
+                    url: "/Bar",
+                    packageKind: .remote,
+                    products: [
+                        ProductDescription(name: "BarLib", targets: ["BarLib"])
+                    ],
+                    targets: [
+                        TargetDescription(name: "BarLib"),
+                    ]),
+                Manifest.createV4Manifest(
+                    name: "Biz",
+                    path: "/BizPath",
+                    url: "/BizPath",
+                    version: "1.2.3",
+                    packageKind: .remote,
+                    products: [
+                        ProductDescription(name: "Biz", targets: ["Biz"])
+                    ],
+                    targets: [
+                        TargetDescription(name: "Biz"),
+                    ]),
+                Manifest.createV4Manifest(
+                    name: "Fiz",
+                    path: "/FizPath",
+                    url: "/FizPath",
+                    version: "1.2.3",
+                    packageKind: .remote,
+                    products: [
+                        ProductDescription(name: "FizLib", targets: ["FizLib"])
+                    ],
+                    targets: [
+                        TargetDescription(name: "FizLib"),
+                    ]),
+            ]
+        )
+
+        DiagnosticsEngineTester(diagnostics) { result in
+            result.checkUnordered(diagnostic: """
+                dependency 'BarLib' in target 'Foo' requires explicit declaration; reference the package in the target \
+                dependency with '.product(name: "BarLib", package: "Bar")'
+                """, behavior: .error, location: "'Foo' /Foo")
+            result.checkUnordered(diagnostic: """
+                dependency 'Biz' in target 'Foo' requires explicit declaration; provide the name of the package \
+                dependency with '.package(name: "Biz", url: "/BizPath", .exact("1.2.3"))'
+                """, behavior: .error, location: "'Foo' /Foo")
+            result.checkUnordered(diagnostic: """
+                dependency 'FizLib' in target 'Foo' requires explicit declaration; reference the package in the target \
+                dependency with '.product(name: "FizLib", package: "FizPath")' and provide the name of the package \
```

Comment:
Shouldn't this say "Fiz" and not "FizPath" in the package name?

Reply:
Also, the name label is not present in `.product`, i.e., it's `.product("productname", package: "...")`

Reply:
Thanks! Fixed.